lib/Sema/CSBindings.cpp

```diff
@@ -1065,6 +1065,14 @@ bool BindingSet::favoredOverDisjunction(Constraint *disjunction) const {
   return !involvesTypeVariables();
 }
 
+bool BindingSet::favoredOverConjunction(Constraint *conjunction) const {
+  if (CS.shouldAttemptFixes() && isHole()) {
+    if (forClosureResult() || forGenericParameter())
```

Comment:
Why is this important in diagnostic mode?

Reply:
No idea, I'm just preserving the original logic :)

Reply:
I should be trying to remember then :)